### PR TITLE
Windows: rewrite paths to configure

### LINKED ISSUE #7649: cabal-install-3.6 fails to install network-3.1.2.2 in windows
* but cabal-install-3.4.0.0 is able to build it
* logs:

```
Configuring network-3.1.2.2...
configure: WARNING: unrecognized options: --with-compiler
checking build system type... x86_64-pc-msys
checking host system type... x86_64-pc-msys
checking for gcc... D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe
checking whether the C compiler works... yes
checking for C compiler default output file name... a.exe
checking for suffix of executables... .exe
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe accepts -g... yes
checking for D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe option to accept ISO C89... none needed
checking for an ANSI C-conforming const... yes
checking how to run the C preprocessor... D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe -E
checking for grep that handles long lines and -e... configure: error: no acceptable grep could be found in /d/ghcup/bin:/c/Users/atrey/AppData/Roaming/cabal/bin:/usr/bin:/mingw64/bin:/d/cabal/store/ghc-8.10.7/hsc2hs-0.68.7-640ce3a674b3ce528257d431f5e15fc40c5ab1a7/bin:/c/Program Files (x86)/Common Files/Oracle/Java/javapath:/c/WINDOWS/system32:/c/WINDOWS:/c/WINDOWS/System32/Wbem:/c/WINDOWS/System32/WindowsPowerShell/v1.0:/c/WINDOWS/System32/OpenSSH:/d/dev/app/git/cmd:/c/Users/atrey/AppData/Local/Microsoft/WindowsApps:/d/bin:/d/dev/ws/haskell/stack:/d/dev/app/emacs-26.2-x86_64/bin:/d/dev/app/Git/bin:/d/dev/app/curl_7_53_1_openssl_nghttp2_x64/bin:/d/dev/app/diffutils-2.8.7-1-bin/bin:/d/dev/app/grep-2.5.4-bin/bin:/d/dev/app/chocolatey/bin:/d/dev/app/vscode/bin:/d/Program Files/Java/jdk1.8.0_231/bin:/d/Program Files/7-Zip:/d/dev/app/node-v12.14.1-win-x64:/d/dev/ws/node.js:/d/dev/app/apache-maven-3.3.9/bin:/d/Program Files/PostgreSQL/11/bin:/d/bin/hls:/d/dev/app/python39/Scripts:/d/dev/app/python39:/d/ghcup/bin:/d/bin/cabal;/usr/xpg4/bin
cabal-3.6.0.0.exe: Failed to build network-3.1.2.2 (which is required by
exe:cabal from cabal-install-3.7.0.0). The failure occurred during the
configure step.
```

with cabal-3.4.0.0

```
Configuring network-3.1.2.2...
configure: WARNING: unrecognized options: --with-compiler
checking build system type... x86_64-pc-msys
checking host system type... x86_64-pc-msys
checking for gcc... D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe
checking whether the C compiler works... yes
checking for C compiler default output file name... a.exe
checking for suffix of executables... .exe
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe accepts -g... yes
checking for D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe option to accept ISO C89... none needed
checking for an ANSI C-conforming const... yes
checking how to run the C preprocessor... D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
....
```

Could it be related with #7510 (which i tested myself 🤦)?

### PR DESCRIPTION
**Pull request by Mistuke**
Should fix #7649 but I don't have msys2 on this machine so other than building Cabal I can't test until I get home.

@jneira @Mikolaj , @jneira could you test this until then?

---
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: Cabal/src/Distribution/Simple.hs [hunk 1]
```diff
@@ -716,6 +718,19 @@ runConfigureScript verbosity backwardsCompatHack flags lbi = do
...
 
+-- | Convert Windows path to Unix ones
+toUnix :: String -> String
+#ifdef mingw32_HOST_OS
+toUnix s = let tmp = normalise s
+               (l:_, rest) = case splitDrive tmp of
+                               ([], x) -> ("/", x)
+                               x       -> x
+               parts = splitDirectories rest
+           in '/' : l : intercalate "/" parts
```
Comment (Mikolaj) [inline Cabal/src/Distribution/Simple.hs:729]: Is there or should there be '/' between `l` and the rest?

**Comment by Mikolaj** [comment]
I don't trust my understanding of this toolset any more, but it generally looks good. Let's remember to revert the failed fix first (#7651) and then this PR can be a tiny bit smaller.

File: Cabal/src/Distribution/Simple.hs (hunk 1 above)
Comment (Mistuke) [reply Cabal/src/Distribution/Simple.hs]: good catch, Just fixed it and the build warning.

**Comment by Mistuke** [comment]
> I don't trust my understanding of this toolset any more, but it generally looks good. Let's remember to revert the failed fix first (#7651) and then this PR can be a tiny bit smaller.

So the idea is to convert `sh C:\/foo/bar/configure` that `cabal` generates into `sh /C/foo/bar/configure`.  so we avoid the `:` in the path.

**Comment by Mikolaj** [comment]
I've merged #7651 that reverts the old fix, hence the conflict, but the conflicting bit can just be removed, it's already applied.

**Comment by Mikolaj** [comment]
Oh, I missed that `{-# LANGUAGE CPP #-}` at the top should not be reverted (so should be added here).

**Comment by Mistuke** [comment]
> Oh, I missed that `{-# LANGUAGE CPP #-}` at the top should not be reverted (so should be added here).

Ah, that's why! lol

**Review by Mikolaj — APPROVED** [review]
Looks good, but we need tests with scripts generated with various versions of autoconf to make sure the layers of hacks around all this don't intervene.

**Comment by jneira** [comment]
I've tested it in my windows 10:

```
PS D:\dev\ws\haskell\issues> cabal get network
Unpacking to network-3.1.2.2\
PS D:\dev\ws\haskell\issues> cd .\network-3.1.2.2\
PS D:\dev\ws\haskell\issues\network-3.1.2.2> D:\bin\cabal.exe build                                 Resolving dependencies...
Build profile: -w ghc-8.10.7 -O1
In order, the following will be built (use -v for more details):
 - network-3.1.2.2 (lib:network) (first run)
Configuring network-3.1.2.2...
configure: WARNING: unrecognized options: --with-compiler
checking build system type... x86_64-pc-msys
checking host system type... x86_64-pc-msys
checking for gcc... D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe
checking whether the C compiler works... yes
checking for C compiler default output file name... a.exe
checking for suffix of executables... .exe
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe accepts -g... yes
checking for D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe option to accept ISO C89... none needed
checking for an ANSI C-conforming const... yes
checking how to run the C preprocessor... D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking for stdlib.h... (cached) yes
checking for unistd.h... (cached) yes
checking for sys/types.h... (cached) yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking sys/uio.h usability... no
checking sys/uio.h presence... no
checking for sys/uio.h... no
checking sys/socket.h usability... no
checking sys/socket.h presence... no
checking for sys/socket.h... no
checking netinet/in.h usability... no
checking netinet/in.h presence... no
checking for netinet/in.h... no
checking netinet/tcp.h usability... no
checking netinet/tcp.h presence... no
checking for netinet/tcp.h... no
checking sys/un.h usability... no
checking sys/un.h presence... no
checking for sys/un.h... no
checking arpa/inet.h usability... no
checking arpa/inet.h presence... no
checking for arpa/inet.h... no
checking netdb.h usability... no
checking netdb.h presence... no
checking for netdb.h... no
checking net/if.h usability... no
checking net/if.h presence... no
checking for net/if.h... no
checking netioapi.h usability... yes
checking netioapi.h presence... yes
checking for netioapi.h... yes
checking for struct ucred... no
checking for gai_strerror... no
checking for gethostent... no
checking for accept4... no
checking for getpeereid... no
checking whether AI_ADDRCONFIG is declared... yes
checking whether AI_ALL is declared... yes
checking whether AI_NUMERICSERV is declared... yes
checking whether AI_V4MAPPED is declared... yes
checking whether IPV6_V6ONLY is declared... yes
checking whether IPPROTO_IP is declared... yes
checking whether IPPROTO_TCP is declared... yes
checking whether IPPROTO_IPV6 is declared... yes
checking whether SO_PEERCRED is declared... no
checking for struct msghdr.msg_control... no
checking for struct msghdr.msg_accrights... no
checking for struct sockaddr.sa_len... no
configure: creating ./network.buildinfo
configure: creating ./config.status
config.status: creating include/HsNetworkConfig.h
configure: WARNING: unrecognized options: --with-compiler
Preprocessing library for network-3.1.2.2..
Building library for network-3.1.2.2..
[ 1 of 28] Compiling Network.Socket.Imports ( Network\Socket\Imports.hs, D:\dev\ws\haskell\issues\network-3.1.2.2\dist-newstyle\build\x86_64-windows\ghc-8.10.7\network-3.1.2.2\build\Network\Socket\Imports.o )
...................................
[28 of 28] Compiling Network.Socket.ByteString.Lazy ( Network\Socket\ByteString\Lazy.hs, D:\dev\ws\haskell\issues\network-3.1.2.2\dist-newstyle\build\x86_64-windows\ghc-8.10.7\network-3.1.2.2\build\Network\Socket\ByteString\Lazy.o )
In file included from D:\ghcup\ghc\8.10.7\lib/include/HsFFI.h:20,

                 from cbits\asyncAccept.c:6:0: error:
D:\ghcup\ghc\8.10.7\lib/include/stg/Types.h:26:9: warning: #warning "Mismatch between __USE_MINGW_ANSI_STDIO definitions. If using Rts.h make sure it is the first header included." [-Wcpp]
   26 | #       warning "Mismatch between __USE_MINGW_ANSI_STDIO definitions. \
      |         ^~~~~~~
In file included from D:\ghcup\ghc\8.10.7\lib/include/HsFFI.h:20,

                 from cbits\initWinSock.c:2:0: error:
D:\ghcup\ghc\8.10.7\lib/include/stg/Types.h:26:9: warning: #warning "Mismatch between __USE_MINGW_ANSI_STDIO definitions. If using Rts.h make sure it is the first header included." [-Wcpp]
   26 | #       warning "Mismatch between __USE_MINGW_ANSI_STDIO definitions. \
      |         ^~~~~~~
In file included from D:\ghcup\ghc\8.10.7\lib/include/HsFFI.h:20,

                 from cbits\winSockErr.c:2:0: error:
D:\ghcup\ghc\8.10.7\lib/include/stg/Types.h:26:9: warning: #warning "Mismatch between __USE_MINGW_ANSI_STDIO definitions. If using Rts.h make sure it is the first header included." [-Wcpp]
   26 | #       warning "Mismatch between __USE_MINGW_ANSI_STDIO definitions. \
      |         ^~~~~~~
```

previously i checked the same steps with the cabal-3.6.0.0 release version and it failed 

Otoh the build generating the config files with autoconf-2.71 in windows also works:

```
atrey@FNTSY MINGW64 /d/dev/ws/haskell/issues
$ cabal get network
Unpacking to network-3.1.2.2\

atrey@FNTSY MINGW64 /d/dev/ws/haskell/issues
$ cd network-3.1.2.2

atrey@FNTSY MINGW64 /d/dev/ws/haskell/issues/network-3.1.2.2
$ rm configure

atrey@FNTSY MINGW64 /d/dev/ws/haskell/issues/network-3.1.2.2
$ rm config.*

atrey@FNTSY MINGW64 /d/dev/ws/haskell/issues/network-3.1.2.2
$ ls
CHANGELOG.md  Network    Setup.hs  configure.ac  include     network.cabal
LICENSE       README.md  cbits     examples      install-sh  tests

atrey@FNTSY MINGW64 /d/dev/ws/haskell/issues/network-3.1.2.2
$ autoreconf --version
autoreconf (GNU Autoconf) 2.71
............

atrey@FNTSY MINGW64 /d/dev/ws/haskell/issues/network-3.1.2.2
$ autoreconf -i

atrey@FNTSY MINGW64 /d/dev/ws/haskell/issues/network-3.1.2.2
$ ls
CHANGELOG.md  README.md       cbits         configure     include        tests
LICENSE       Setup.hs        config.guess  configure.ac  install-sh
Network       autom4te.cache  config.sub    examples      network.cabal

atrey@FNTSY MINGW64 /d/dev/ws/haskell/issues/network-3.1.2.2
$ /d/bin/cabal.exe build
Resolving dependencies...
Build profile: -w ghc-8.10.7 -O1
In order, the following will be built (use -v for more details):
 - network-3.1.2.2 (lib:network) (first run)
Configuring network-3.1.2.2...
configure: WARNING: unrecognized options: --with-compiler
configure: loading site script D:/ghcup/msys64/etc/config.site
checking build system type... x86_64-w64-mingw32
checking host system type... x86_64-w64-mingw32
checking for gcc... D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe
checking whether the C compiler works... yes
checking for C compiler default output file name... a.exe
checking for suffix of executables... .exe
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe accepts -g... yes
checking for D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe option to enable C11 features... none needed
checking for an ANSI C-conforming const... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for limits.h... yes
checking for stdlib.h... (cached) yes
checking for unistd.h... (cached) yes
checking for sys/types.h... (cached) yes
checking for fcntl.h... yes
checking for sys/uio.h... no
checking for sys/socket.h... no
checking for netinet/in.h... no
checking for netinet/tcp.h... no
checking for sys/un.h... no
checking for arpa/inet.h... no
checking for netdb.h... no
checking for net/if.h... no
checking for netioapi.h... yes
checking for struct ucred... no
checking for gai_strerror... no
checking for gethostent... no
checking for accept4... no
checking for getpeereid... no
checking for D:\ghcup\ghc\8.10.7\lib\../mingw/bin\gcc.exe options needed to detect all undeclared functions... none needed
checking whether AI_ADDRCONFIG is declared... yes
checking whether AI_ALL is declared... yes
checking whether AI_NUMERICSERV is declared... yes
checking whether AI_V4MAPPED is declared... yes
checking whether IPV6_V6ONLY is declared... yes
checking whether IPPROTO_IP is declared... yes
checking whether IPPROTO_TCP is declared... yes
checking whether IPPROTO_IPV6 is declared... yes
checking whether SO_PEERCRED is declared... no
checking for struct msghdr.msg_control... no
checking for struct msghdr.msg_accrights... no
checking for struct sockaddr.sa_len... no
configure: creating ./network.buildinfo
configure: creating ./config.status
config.status: creating include/HsNetworkConfig.h
configure: WARNING: unrecognized options: --with-compiler
Preprocessing library for network-3.1.2.2..
........
```

**Comment by Mikolaj** [comment]
@jneira: in the first log, these are only warnings, right? As @Mistuke predicted? All run eventually succeeded?

**Comment by jneira** [comment]
> 
> 
> @jneira: in the first log, these are only warnings, right? As @Mistuke predicted? All run eventually succeeded?

yeah the build succeded, those warning were present before this version

**Comment by jneira** [comment]
`time` and `process` also worked in my end

**Review by jneira — APPROVED** [review]
LGTM, thanks @Mistuke for taking care

**Comment by Mikolaj** [comment]
@Mistuke: we are waiting until you have some breathing space to have a slow, careful, final look before this is merged. No pressure, of course, the whole world can wait. ;D

I mean, this problem evaded quite extensive testing last time, so we are not as keen as before to believe the tests this time. Hence the special weight of your expertise.

**Comment by Mikolaj** [comment]
I've merged #7653 with extra checks that may be useful to test this PR, so let me rebase and see.

**Comment by Mikolaj** [comment]
@mergifyIO rebase

**Comment by Mistuke** [comment]
> @Mistuke: we are waiting until you have some breathing space to have a slow, careful, final look before this is merged. No pressure, of course, the whole world can wait. ;D

Morning! I have some work to finish this morning but will do some manual checks today after work in ~9h.

**Comment by Mistuke** [comment]
@jneira That's for also testing so far.

**Comment by jneira** [comment]
will that works for all the weird paths on windows systems (or maybe not make sense cause they will? i am thinking in shortened ones like `C:\xx~\yyy`

> 
> 
> @jneira That's for also testing so far.

I tried to test it with autoconf-2.69 but i was not able to make it generate `config.sub` and `config.guess`, even using `--force`. so the consequent build fails cause they are not.
I downgraded it with `pacman -R autoconf`, downloading https://repo.msys2.org/msys/sources/autoconf-2.69-5.src.tar.gz and installing it with `pacman -U path/to/autoconf-2.69-5.src.tar.gz`. Maybe i need other version?

**Comment by Mistuke** [comment]
> I tried to test it with autoconf-2.69 but i was not able to make it generate config.sub and config.guess, even using --force. so the consequent build fails cause they are not.

@jneira those files belong to automake, autoconf only reads them not makes them. If you really want to generate new ones you should use `automake -a` after removing them.

**Comment by Mistuke** [comment]
> @jneira: in the first log, these are only warnings, right? As @Mistuke predicted? All run eventually succeeded?

those warnings are a bug in network and are produced by GHC as a sanity check.

**Comment by Mistuke** [comment]
It looks like only the path to configure is being rewritten as expected:
```
"C:\tools\msys64\usr\bin\sh.exe" "/C/Users/xbox/source/repos/network/configure" "--with-compiler=ghc" "--prefix=C:\Users\xbox\AppData\Roaming\cabal" "--bindir=C:\Users\xbox\AppData\Roaming\cabal\bin" "--libdir=C:\Users\xbox\AppData\Roaming\cabal\x86_64-windows-ghc-8.10.3\network-3.2.0.0-inplace" "--libexecdir=C:\Users\xbox\AppData\Roaming\cabal\network-3.2.0.0-inplace\x86_64-windows-ghc-8.10.3\network-3.2.0.0" "--datadir=C:\Users\xbox\AppData\Roaming\cabal\x86_64-windows-ghc-8.10.3\network-3.2.0.0" "--sysconfdir=C:\Users\xbox\AppData\Roaming\cabal\etc" "CC=C:\PROGRA~3\CHOCOL~1\lib\ghc\tools\GHC-81~1.3\lib\../mingw/bin\gcc.exe"
```

so that's good. 

Building both my local and package network works:
```
Build profile: -w ghc-8.10.3 -O1
In order, the following will be built (use -v for more details):
 - network-3.1.2.2 (lib:network) (requires build)
Starting     network-3.1.2.2 (all, legacy fallback)
Building     network-3.1.2.2 (all, legacy fallback)
Installing   network-3.1.2.2 (all, legacy fallback)
Completed    network-3.1.2.2 (all, legacy fallback)
```

Looking at the log file:

```
It was created by Haskell network package configure 3.1.2.1, which was
generated by GNU Autoconf 2.71.  Invocation command line was

  $ /C/Users/xbox/source/repos/network/configure --with-compiler=ghc '--prefix=C:\Users\xbox\AppData\Roaming\cabal' '--bindir=C:\Users\xbox\AppData\Roaming\cabal\bin' '--libdir=C:\Users\xbox\AppData\Roaming\cabal\x86_64-windows-ghc-8.10.3\network-3.2.0.0-inplace' '--libexecdir=C:\Users\xbox\AppData\Roaming\cabal\network-3.2.0.0-inplace\x86_64-windows-ghc-8.10.3\network-3.2.0.0' '--datadir=C:\Users\xbox\AppData\Roaming\cabal\x86_64-windows-ghc-8.10.3\network-3.2.0.0' '--sysconfdir=C:\Users\xbox\AppData\Roaming\cabal\etc' 'CC=C:\PROGRA~3\CHOCOL~1\lib\ghc\tools\GHC-81~1.3\lib\../mingw/bin\gcc.exe'
....
configure:2408: looking for aux files: config.guess config.sub
configure:2421:  trying /C/Users/xbox/source/repos/network/
configure:2450:   /C/Users/xbox/source/repos/network/config.guess found
configure:2450:   /C/Users/xbox/source/repos/network/config.sub found
configure:2628: checking build system type
configure:2643: result: x86_64-pc-msys
configure:2663: checking host system type
configure:2677: result: x86_64-pc-msys
```
and nothing else uses the new paths, Everything else still uses `C:\/` as before.

**Comment by Mistuke** [comment]
WIth packaged network
```
It was created by Haskell network package configure 3.1.2.1, which was
generated by GNU Autoconf 2.69.  Invocation command line was

  $ /C/Users/xbox/AppData/Local/Temp/network-3.1.2.2/configure --with-compiler=ghc --prefix=C:\Users\xbox\AppData\Roaming\cabal --bindir=C:\Users\xbox\AppData\Roaming\cabal\bin --libdir=C:\Users\xbox\AppData\Roaming\cabal\x86_64-windows-ghc-8.10.3\network-3.1.2.2-inplace --libexecdir=C:\Users\xbox\AppData\Roaming\cabal\network-3.1.2.2-inplace\x86_64-windows-ghc-8.10.3\network-3.1.2.2 --datadir=C:\Users\xbox\AppData\Roaming\cabal\x86_64-windows-ghc-8.10.3\network-3.1.2.2 --sysconfdir=C:\Users\xbox\AppData\Roaming\cabal\etc CC=C:\PROGRA~3\CHOCOL~1\lib\ghc\tools\GHC-81~1.3\lib\../mingw/bin\gcc.exe
```
`PATH` are expanded early, but is sane..
```
PATH: /mingw64/bin
PATH: /c/Users/xbox/AppData/Roaming/cabal/bin
PATH: /usr/bin
PATH: /c/Users/xbox/AppData/Roaming/cabal/store/ghc-8.10.3/hsc2hs-0.68.7-c751d9ef1654577188b138cf30f3aaffa4c50115/bin
PATH: /c/Program Files (x86)/IntelSWTools/compilers_and_libraries_2020.1.216/windows/mpi/intel64/bin
PATH: /c/Program Files (x86)/Common Files/Intel/Shared Libraries/redist/intel64_win/mpirt
PATH: /c/Program Files (x86)/Common Files/Intel/Shared Libraries/redist/intel64_win/compiler
PATH: /c/WINDOWS/system32
PATH: /c/WINDOWS
...
```
and detected as
```
configure:2353: checking build system type
configure:2367: result: x86_64-pc-msys
configure:2387: checking host system type
configure:2400: result: x86_64-pc-msys
configure:2475: checking for gcc
```
and the new path isn't used by anything else.

**Comment by Mistuke** [comment]
Does not work using networked paths
```
cc1.exe: error: /mingw64/include: Invalid argument
`gcc.exe' failed in phase `C Compiler'. (Exit code: 1)
```
but old cabal never did either
```
Project settings changed, reconfiguring...
creating \\exia\e$\buggy\network-3.1.2.2\dist-newstyle
creating \\exia\e$\buggy\network-3.1.2.2\dist-newstyle\cache
\\pinky\e$\buggy\network-3.1.2.2\.\network.cabal: openBinaryFile: does not exist (No such file or directory)
```
and that's because GCC doesn't support UNC paths. neither cases ever get to configure.

**Comment by Mistuke** [comment]
to support UNC path this case needs to be added:

```
Windows: splitDrive "\\\\shared\\test" == ("\\\\shared\\","test")
```
where `\\\\shared\\test` needs to be mapped to `//shared/test`. However I consider this a new feature tbh. but can add the case if desired but will have no way of testing it as we die before configure.

**Comment by Mikolaj** [comment]
Why do we die before configure?

**Comment by jneira** [comment]
> 
> 
> > I tried to test it with autoconf-2.69 but i was not able to make it generate config.sub and config.guess, even using --force. so the consequent build fails cause they are not.
> 
> @jneira those files belong to automake, autoconf only reads them not makes them. If you really want to generate new ones you should use `automake -a` after removing them.

makes sense but it seems 2.71 is able to generate them standalone for me

**Comment by jneira** [comment]
> Does not work using networked paths

What about shortened paths like `C:\xxx~\yyy`, i guess they should staring at the code

**Comment by Mistuke** [comment]
> 
> 
> Why do we die before configure?

because GCC does not support UNC paths, and depending on which ghc version is used to compile cabal, neither does cabal.

**Comment by Mistuke** [comment]
> 
> 
> > Does not work using networked paths
> 
> What about shortened paths like `C:\xxx~\yyy`, i guess they should staring at the code

Dos shortnames are a file system concept. They hold no meaning in anything but NTFS, as such they are handled by the file system driver and not a user mode application. To userland (and msys2) there's no difference when they are used or not so they work just fine.

**Comment by Mistuke** [comment]
> 
> 
> > > I tried to test it with autoconf-2.69 but i was not able to make it generate config.sub and config.guess, even using --force. so the consequent build fails cause they are not.
> > 
> > 
> > @jneira those files belong to automake, autoconf only reads them not makes them. If you really want to generate new ones you should use `automake -a` after removing them.
> 
> makes sense but it seems 2.71 is able to generate them standalone for me

Yes but that's a change in autoconf 2.70 https://github.com/autotools-mirror/autoconf/blob/master/NEWS#L105

**Comment by Mistuke** [comment]
in any case, manually calling `sh` with the `path` cabal would use if it didn't die earlier works.
```
C:\tools\msys64\usr\bin\sh.exe --login "\\pinky\e$\buggy\network/configure" "--with-compiler=ghc" "--prefix=..
configure: loading site script /etc/config.site
\\pinky\e$\buggy\network/configure: line 2615: :=;: command not found
checking build system type... x86_64-pc-msys
checking host system type... x86_64-pc-msys
...
```
So should be fine.

**Comment by Mikolaj** [comment]
> > Why do we die before configure?
> 
> because GCC does not support UNC paths, and depending on which ghc version is used to compile cabal, neither does cabal.

In that case, let's wait until it's easier to test and until users request the feature and are willing to test on their own.

**Comment by jneira** [comment]
> Dos shortnames are a file system concept. They hold no meaning in anything but NTFS, as such they are handled by the file system driver and not a user mode application. To userland (and msys2) there's no difference when they are used or not so they work just fine.

Thanks, i might overlook it remembering this issue: https://github.com/haskell/cabal/issues/7170

**Comment by Mistuke** [comment]
> 
> 
> > Dos shortnames are a file system concept. They hold no meaning in anything but NTFS, as such they are handled by the file system driver and not a user mode application. To userland (and msys2) there's no difference when they are used or not so they work just fine.
> 
> Thanks, i might overlook it remembering this issue: #7170

That's unrelated to this. There's no configuring going on there and the post contains too few information to tell what's going on. 
It's likely some other tool or part of cabal thinks it's invalid. https://ss64.com/nt/syntax-filenames.html

Also these things come with a significant performance cost (which is why I and a lot of people have them disabled everywhere):

```
In any directory containing more than 25,000 files there is a significant performance cost ~ 6,000 files per hour vs 2 million files per hour.
```

**Comment by jneira** [comment]
> 
> 
> > > Dos shortnames are a file system concept. They hold no meaning in anything but NTFS, as such they are handled by the file system driver and not a user mode application. To userland (and msys2) there's no difference when they are used or not so they work just fine.
> > 
> > 
> > Thanks, i might overlook it remembering this issue: #7170
> 
> That's unrelated to this. There's no configuring going on there and the post contains too few information to tell what's going on.
> It's likely some other tool or part of cabal thinks it's invalid. https://ss64.com/nt/syntax-filenames.html
> 
> Also these things come with a significant performance cost (which is why I and a lot of people have them disabled everywhere):
> 
> ```
> In any directory containing more than 25,000 files there is a significant performance cost ~ 6,000 files per hour vs 2 million files per hour.
> ```

Thanks for share your knowledge and your patience :slightly_smiling_face: 
Will try to reproduce the issue to provide more info

**Comment by Mistuke** [comment]
> Thanks for share your knowledge and your patience 🙂
> Will try to reproduce the issue to provide more info

That issue just looks like either cabal is not given the short pathname at all by the shell or something is expanding it to the full path.

I'll be honest, not really sure it's worth fixing. I think focus should be to just enable winio and move on. Dos shortnames are legacy and not supported by newer windows filesystems anyway.

**Comment by Mikolaj** [comment]
CI on master is now fixed, so let me rebase.

**Comment by Mikolaj** [comment]
@mergifyIO rebase

**Comment by Mikolaj** [comment]
Oh, dear, I forgot the PR that fixes the last bit of master CI doesn't have enough reviews and so is not merged. So, it's going to fail.

**Comment by jneira** [comment]
#7659 is merged so rebasing
@MergifyIO rebase

**Comment by jneira** [comment]
@MergifyIO backport 3.6

**Comment by Mikolaj** [comment]
Well done. Full green CI, finally, including the dogfooding tests that catch the original bug in cabal 3.6 (though not the new autoconf problems that were present in 3.4, because the new autoconf configs are not yet present in packages on Hackage, I think).

**Comment by Mikolaj** [comment]
As soon as #7664 gets merged to 3.6 by mergifyIO, we can backport this again to 3.6 and it should merge correctly and pass CI fine this time.

This will add the last missing piece in the back/forward-porting effort related to 3.6.2 and CI failures:

```
PRs on master (from old to new):
1 #7651 v
2 #7653 v
3 #7650 v
4 #7658 v forwardport of #7657
5 #7659 v
6 #7652

corresponding PR backports (some extended) on 3.6:
2 #7653
4 #7657 forward-ported to #7658
5 #7659
3 #7650
  #7665   3.6-exclusive CI fix
1 #7651
```

**Comment by jneira** [comment]
Wow, many thanks for this tour the force and share the info

**Comment by Mikolaj** [comment]
I'm not working today, but let me just repeat what @jneira did 18 hours ago and for what 3.6 is now finally ready:
@Mergifyio backport 3.6

**Comment by Mikolaj** [comment]
Huh, actually mergifyIO mentions the same old PR instead of creating a new one. So I guess, this needs to be backported manually after all. See you on Monday!

**Pull request merged**